metrics/stmt.current/current.go

```diff
@@ -0,0 +1,168 @@
+package stmt
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/cashapp/blip"
+)
+
+const (
+	DOMAIN = "stmt.current"
+	query  = `SELECT TIMER_WAIT FROM performance_schema.events_statements_current 
+		WHERE END_EVENT_ID IS NULL 
+		AND EVENT_NAME NOT LIKE ('statement/com/Binlog%')`
+	OPT_THRESHOLD = "threshold"
+)
+
+type currentMetrics struct {
+	slowest   bool
+	slow      bool
+	threshold float64
+}
+
+// Stmt collects metrics for the stmt domain.
+// The source is performance_schema.events_statements_current.
+type Current struct {
+	db      *sql.DB
+	atLevel map[string]currentMetrics
+}
+
+var _ blip.Collector = &Current{}
+
+func NewCurrent(db *sql.DB) *Current {
+	return &Current{
+		db:      db,
+		atLevel: map[string]currentMetrics{},
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (c *Current) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (c *Current) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Statement metrics",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_THRESHOLD: {
+				Name:    OPT_THRESHOLD,
+				Desc:    "The length of time (in microseconds) that a query must be active to be considered slow",
+				Default: "30000000",
```

Comment:
Let's use a duration string and default "1.0s" and minimum value 1 us. Reason being: I was trying to think of a reasonable default (30s is way to long imho)... but we'll never know what users think is "long". So 1.0 is a nice "round" starting point (1s queries _are_ pretty slow in a normal-ish MySQL db, I'd say). Then since we don't know what users will want, the millisecond format is too messy; duration strings are easy to write for any range of values.